Raygun.Druid4Net/Query/Results/SegmentMetadataResult.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+
+namespace Raygun.Druid4Net
+{
+  public class SegmentMetadataResult : List<SegmentMetadataResultItem>
+  {
+  }
+
+  public class SegmentMetadataResultItem
+  {
+    public string Id { get; set; }
+    
+    public IEnumerable<string> Intervals { get; set; }
+    
+    public Dictionary<string, SegmentMetadataColumnDescriptor> Columns { get; set; }
+    
+    public Dictionary<string, SegmentMetadataAggregatorDescriptor> Aggregators { get; set; }
+    
+    public IGranularitySpec QueryGranularity { get; set; }
+    
+    public int Size { get; set; }
+    
+    public int NumRows { get; set; }
+  }
+
+  public class SegmentMetadataColumnDescriptor
+  {
+    public SegmentMetadataColumnType Type { get; set; }

```

Comment:
Does this support this scenario as per the docs?
> Metric columns will have type FLOAT, DOUBLE, or LONG, or the name of the underlying complex type such as hyperUnique in case of COMPLEX metric.

Specifically where the type could be the name of the complex type, e.g. hyperUnique?

Reply:
Ah good spot, I'd misread the documentation. In other areas of Druid, such as aggregators, they just have a catch all `COMPLEX` value in their enum and I assumed it was the same here. Do you have any suggestions for how we could handle this? We could simply use a string?

Reply:
Yeah, I think the only option here is to make it a string.

Reply:
Just pushed up a change (21efb3e2b69739919cf45b09ba364b5a966757a5) to cater for this example.